### DISCUSSION TITLE
Error: Class not found when running custom command in Winter CMS

### DISCUSSION THREAD

**Discussion by 01Kuzma** [wintercms/winter#941] [Q&A] [answered]
Hello!
With a colleague, we are trying to write an app which is working with DB and the Trask manager (CRON).

The Winter CMS version is 1.2.3 installed in Windows with a web installer.

Firstly we have created a Plugin which adds the entries to DB. That's OK, but the Task Manager's part is not working.
We have registered the command as follows in the `Plugin.php`:

```
<?php namespace Kuzma\Employees;

use Kuzma\Employees\Console\Commands\CheckDatesCommand;
use Illuminate\Support\Facades\Event; 
use Illuminate\Console\Scheduling\Schedule;
use System\Classes\PluginBase;

class Plugin extends PluginBase
{
    public function registerComponents()
    {
    }

    public function registerSettings()
    {
    }
	
    public function boot()
    {     

        Event::listen('scheduler.beforeSchedule', function (Schedule $schedule) {
            $schedule->command('employees:check-dates')->daily();
        });
    }	
	
    public function register()
    {
        $this->registerConsoleCommand('employees:check-dates', CheckDatesCommand::class);
    }	
}

```

The `CheckDatesCommand.php` is placed here: `C:\laragon\www\project_name\plugins\kuzma\employees\classes\console\commands\`

`CheckDatesCommand.php`:

```
<?php
namespace Kuzma\Employees\Console\Commands;

use Illuminate\Console\Command;
use Kuzma\Employees\Models\Employee;

class CheckDatesCommand extends Command
{
    protected $signature = 'employees:check-dates';

    protected $description = 'Check dates and update fields accordingly';

    public function handle()
    {
         //...
        }

        $this->info('Dates checked and fields updated successfully.');
    }

    private function sendReminderEmail(Employee $employee)
    {

    }
}

```
Once the command `php artisan employees:check-dates` run, we have these errors:

```
C:\laragon\www\project_name>php artisan employees:check-dates

In Container.php line 891:

  Target class [Kuzma\Employees\Console\Commands\CheckDatesCommand] does not exist.


In Container.php line 889:

  Class "Kuzma\Employees\Console\Commands\CheckDatesCommand" does not exist
 ```

What's missing? Thank you!

**Comment by bennothommo** [comment]
Hi @01Kuzma, it might be the file path you are using. It should be `C:\laragon\www\project_name\plugins\kuzma\employees\console\commands\` (ie. not including the `classes` subdirectory) in order for the class loader to find the correct file based on the namespace you are using for the class.

**Comment by 01Kuzma** [comment]
Thank you for the answer @bennothommo!
I've excluded the `classes` folder, now the structure looks: 
`C:\laragon\www\project_name\plugins\kuzma\employees\console\commands\`

But the errors are the same :(

```
C:\laragon\www\project_name>php artisan employees:check-dates

In Container.php line 891:

  Target class [Kuzma\Employees\Console\Commands\CheckDatesCommand] does not exist.


In Container.php line 889:

  Class "Kuzma\Employees\Console\Commands\CheckDatesCommand" does not exist
```

Also, if I ran different command, for example: `php artisan cache:clear`
The error is the same... From that plugin

**Comment by 01Kuzma** [comment, marked as answer]
It seems, that this was the issue `namespace Kuzma\Employees\Console\Commands;` :)

it should look like:
```
<?php
namespace Kuzma\Employees\Console\Commands;

use Winter\Storm\Console\Command;
use Kuzma\Employees\Models\Employee;

class CheckDatesCommand extends Command
{
}
```
